python/source/org/libreoffice/ide/eclipse/python/Language.java

```diff
@@ -94,31 +94,55 @@ public void connectDebuggerToOffice(IUnoidlProject pPrj, ILaunch pLaunch, IPath
-            //FIXME implement some kind of port pickup/retry mechanism in case the default port is already used.
-            argMap.put("port", DEFAULT_JAVA_DEBUG_PORT);
-
-            connector.connect(argMap, pMonitor, pLaunch);
+            if (PydevRemoteDebuggerServer.isRunning()) {
+                pPrj.getOOo().runOffice(pPrj, pLaunch, pUserInstallation, new NullExtraOptionsProvider(), pMonitor);
+            } else {
+                /* This allows to start the server asynchronously which will give 'Invalid thread access' Exception
+                   if started from this thread which has been called by DebugUITools.launch(...)*/
+                Display.getDefault().asyncExec(new debugSeverStart(pPrj, pLaunch, pUserInstallation, pMonitor));
+            }
 
-            pPrj.getOOo().runOffice(pPrj, pLaunch, pUserInstallation,
-                new PythonDebugExtraOptionsProvider(DEFAULT_JAVA_DEBUG_PORT), pMonitor);
         } catch (Exception e) {
-            PluginLogger.error("Could not start remote debugger.", e);
+            PluginLogger.error("Could not go for Debugging Mode.", e);
         }
     }
 
     @Override
     public void configureSourceLocator(ILaunchConfigurationWorkingCopy pConfiguration) throws CoreException {
         String projectName = pConfiguration.getAttribute(IOfficeLaunchConstants.PROJECT_NAME, "");
         pConfiguration.setAttribute(ILaunchConfiguration.ATTR_SOURCE_LOCATOR_ID,
-            "org.eclipse.jdt.launching.sourceLocator.JavaSourceLookupDirector");
+            "org.eclipse.jd/t.launching.sourceLocator.JavaSourceLookupDirector");
         pConfiguration.setAttribute(ISourcePathComputer.ATTR_SOURCE_PATH_COMPUTER_ID,
             "org.eclipse.jdt.launching.sourceLookup.javaSourcePathComputer");
         pConfiguration.setAttribute(IJavaLaunchConfigurationConstants.ATTR_PROJECT_NAME, projectName);
     };
+
+    /**
+     * Thread executing the starting of the server followed by launching the LibreOffice instance for debugging 
+     */
+    private class debugSeverStart implements Runnable {
+
+        private IUnoidlProject mPrj;
+        private ILaunch mLaunch;
+        private IPath mUserInstallation;
+        private IProgressMonitor mMonitor;
+
+        public debugSeverStart(IUnoidlProject pPrj, ILaunch pLaunch, IPath pUserInstallation,
+            IProgressMonitor pMonitor) {
+            mPrj = pPrj;
+            mLaunch = pLaunch;
+            mUserInstallation = pUserInstallation;
+            mMonitor = pMonitor;
+        }
+
+        @Override
+        public void run() {
+            try {
+                PydevRemoteDebuggerServer.startServer();
+                mPrj.getOOo().runOffice(mPrj, mLaunch, mUserInstallation, new NullExtraOptionsProvider(), mMonitor);
+            } catch (Exception e) {
+                PluginLogger.error("Could not start the debug server and start the LibreOffice Instance.", e);
+            }
+        }
+    }
```

Comment:
This allows us to start the PyDev server to be started when user selects the Debug mode, we have to go for create a new thread and execute it in a asynchronus mode since in the same thread it will throw 'Invalid thread access'